### PR TITLE
Zelling Harris transformations

### PR DESCRIPTION
**Pull request by VincTheSecond**
Initial commit of the Harris's transformations.

In addition, several new general features were added (I'm very sorry that these changes do not come with several independent pull requests :-[ )
 - reading of the BZ2-compressed CONNLU files
 - filtering sentences/bundles according to the sentence ID
 - a new read block for reduced CONNLU files
 - a new write block for writing ASCII dependency trees (will be improved soon)

### CODE REVIEW

File: udapi/block/read/conllu.py [hunk 1]
```diff
@@ -86,6 +105,19 @@ def process_document(self, document):
                 logging.debug('No next bundle to process')
                 break
 
+            # Check if all lines have correct number of data fields.
+            # Skip invalid bundle otherwise.
+            raw_bundle_check = True
+            for line in raw_bundle:
+                if re_comment_like.search(line) is not None:
+                    continue
+                if len(line.split('\t')) != len(self.node_attributes):
```
Comment (martinpopel) [inline udapi/block/read/conllu.py]: Loading CoNLL-U is the speed-critical part of code. Here the split('\t') is done twice, so I am afraid it is slower than the previous version. It should be benchmarked, e.g. using the https://github.com/martinpopel/newtreex#benchmark (where "utreex" is an older nickname for this python-udapi). 
As for validity checking: ideally there should be a fast&robust way and slower&strict way of loading.

File: udapi/block/write/asciitree.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+import logging
+import codecs
+import sys
+
+from udapi.core.block import Block
+
+
+def get_number_of_antecedents(node):
```
Comment (martinpopel) [inline udapi/block/write/asciitree.py]: See https://github.com/udapi/udapi-perl/blob/master/lib/Udapi/Block/Write/TextModeTrees.pm
- [ ] add here a link to the Perl implementation as a TODO (parameters, colors, nonprojectivities etc.)
- [ ] I think we should call this block the same in Perl, Java and Python: I think both TextModeTrees and AsciiTrees is ok, with a slight preference to the former.

File: udapi/block/zellig_harris/baseline.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+        node_form = node.form.lower()
+        if self.suffixed_forms:
+            node_form = node_form[:-3]
+
+        parent_form = node.parent.form.lower()
+        if self.suffixed_forms:
+            parent_form = parent_form[:-3]
+
+        # Process node's parent.
+        parent_deprel_orig = node.deprel
```
Comment (martinpopel) [inline udapi/block/zellig_harris/baseline.py]: Shouldn't it be `node.parent.deprel`? In line with the `parent_form = node.parent.form.lower()` above.

File: udapi/block/zellig_harris/baseline.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+                has_preposition = False
+                for subchild in child.children:
+                    if subchild.deprel == 'case':
+                        has_preposition = True
+                        break
+
+                if has_preposition:
+                    print "%s %s_%s" % (node_form, child_form, 'prep')
+
+
```
Comment (martinpopel) [inline udapi/block/zellig_harris/baseline.py]: why empty lines?

File: udapi/block/zellig_harris/baseline.py (hunk 3 above)
Comment (VincTheSecond) [inline udapi/block/zellig_harris/baseline.py]: This is correct.
The relation label between `(parent, node)` is stored in the `node.deprel`. The same as `afun` in the PML trees.

Comment (martinpopel) [inline udapi/block/zellig_harris/baseline.py]: OK. I just wanted to check it is not a typo.
(Deprel is a linguistic relation between node A and its parent B, but technically it is always stored in A, so I would be confused by your code if I see variables `parent_form` and `parent_deprel` where the former is stored in B and the latter in A.)

File: udapi/block/write/asciitree.py (hunk 2 above)
Comment (VincTheSecond) [inline udapi/block/write/asciitree.py]: I renamed the block and add the notice about the Perl implementation.

File: udapi/block/read/conllu.py (hunk 1 above)
Comment (VincTheSecond) [inline udapi/block/read/conllu.py]: Should I add a special parameter `strict` which will enable more controlled processing?

Comment (VincTheSecond) [inline udapi/block/read/conllu.py]: To do the benchmark -- is it enough to run `time udapi ...` with and without multiple line splitting?

Comment (VincTheSecond) [inline udapi/block/read/conllu.py]: Processing time of the `ud-treebanks-v1.4/UD_Czech/cs-ud-train.conllu` file (68495 bundles):
```
master                       : 1m2.148s
zellig_haris                 : 1m2.184s
zellig_haris + strict option : 1m5.071s
```

**Pull request merged**